src/main/java/com/epam/training/designpatterns/fastfoodrestaurant/RestaurantSimulator.java

```diff
@@ -0,0 +1,73 @@
+package com.epam.training.designpatterns.fastfoodrestaurant;
+
+import java.util.Random;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import com.epam.training.designpatterns.fastfoodrestaurant.entities.Client;
+import com.epam.training.designpatterns.fastfoodrestaurant.entities.Order;
+import com.epam.training.designpatterns.fastfoodrestaurant.food.Chips;
+import com.epam.training.designpatterns.fastfoodrestaurant.food.Condiment;
+import com.epam.training.designpatterns.fastfoodrestaurant.food.Food;
+import com.epam.training.designpatterns.fastfoodrestaurant.food.Hotdog;
+import com.epam.training.designpatterns.fastfoodrestaurant.food.Ketchup;
+import com.epam.training.designpatterns.fastfoodrestaurant.food.Mustard;
+import com.epam.training.designpatterns.fastfoodrestaurant.strategies.AmericanStyleCookingStrategy;
+import com.epam.training.designpatterns.fastfoodrestaurant.strategies.CookingStrategy;
+import com.epam.training.designpatterns.fastfoodrestaurant.strategies.JapaneseStyleCookingStrategy;
+import com.epam.training.designpatterns.fastfoodrestaurant.workstations.Chef;
+import com.epam.training.designpatterns.fastfoodrestaurant.workstations.DeliveryQueue;
+import com.epam.training.designpatterns.fastfoodrestaurant.workstations.OrderQueue;
+import com.epam.training.designpatterns.fastfoodrestaurant.workstations.Waiter;
+
+public class RestaurantSimulator {
+
+	public static final Random random = new Random();
+	
+	private OrderQueue orderQueue = new OrderQueue();
+	private DeliveryQueue deliveryQueue = new DeliveryQueue();
+	private Chef chef = new Chef(orderQueue, deliveryQueue, new AmericanStyleCookingStrategy());
+	private Waiter waiter = new Waiter(orderQueue, deliveryQueue);
+	
+	
+	public static void main(String[] args) throws InterruptedException {
+		RestaurantSimulator restaurant = new RestaurantSimulator();
+		restaurant.startSimulation(20, 4000);
+	}
+	
+	
+	public void startSimulation(int maxNumberOfClients, int maxSpawnTime) throws InterruptedException {
+		ExecutorService executor = Executors.newCachedThreadPool();
+		for (int i = 0; i < maxNumberOfClients; ++i) {
+			executor.submit(new SimulatedClient());
+			Thread.sleep(maxSpawnTime);
+		}
+	}
+	
+	private class SimulatedClient implements Runnable {
+
+		@Override
+		public void run() {
+			Client client = new Client(waiter);
+			Order order = generateRandomOrder(client);
+			client.orderFood(order);
+		}
+		
+		private Order generateRandomOrder(Client client) {
+			Food food = random.nextInt(100) > 50 ? new Chips() : new Hotdog();
```

Comment:
Ezeket a lépéseket légyszi tedd ki minimum private metódusokba, de enterprise szinten simán kaphatnának saját factory osztályt is.

Ilyenre gondolok:
```
private Order generateRandomOrder(Client client) {
Food food = createRandomFood();
...
private Food createRandomFood() {
    return random.nextInt(100) > 50 ? new Chips() : new Hotdog();
}
```
vagy akár
```
Food food = randomFoodFactory.create();
...
public class RandomFoodFactory {

    public Food create() {
        return random.nextInt(100) > 50 ? new Chips() : new Hotdog();
    }
}
```
És így tovább a `condiment`, a `priority` és a `cookingStyle` esetében.

A random szám generálása is kimehet private metódusba, illetve akár osztályba is. Ilyenkor persze az összes factory-nak szüksége lesz arra a random számot generáló osztályra és adódik a kérdés, hogy hány példány kell majd belőle. Egy DI framework levenné ezt a gondot a vállunkról, de itt nem használunk.

Legyenek most a private metódusok!

Reply:
Abszolút, ez tök jogos, hiszen így ha kiteszem akkor bármikor kicserélhetem a bennük lévő implementációt, nincs beleégetve a SimulatedClient-be a megvalósítás.